src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FootnoteService.cs

```diff
@@ -70,21 +55,20 @@ public class FootnoteService : IFootnoteService
                 .OnSuccess(_userService.CheckCanUpdateRelease)
                 .OnSuccess(async () =>
                 {
-                    var footnote = (await _context.Footnote.AddAsync(new Footnote
+                    var newFootnoteId = Guid.NewGuid();
+
+                    var footnote = new Footnote
                     {
+                        Id = newFootnoteId,
                         Content = content,
-                        Filters = new List<FilterFootnote>(),
-                        FilterGroups = new List<FilterGroupFootnote>(),
-                        FilterItems = new List<FilterItemFootnote>(),
-                        Indicators = new List<IndicatorFootnote>(),
-                        Subjects = new List<SubjectFootnote>()
-                    })).Entity;
-
-                    CreateSubjectLinks(footnote, subjectIds);
-                    CreateFilterLinks(footnote, filterIds);
-                    CreateFilterGroupLinks(footnote, filterGroupIds);
-                    CreateFilterItemLinks(footnote, filterItemIds);
-                    CreateIndicatorsLinks(footnote, indicatorIds);
+                        Subjects = CreateSubjectLinks(newFootnoteId, subjectIds),
```

Comment:
I was also considering removing the lookups of the various link entities before creating the links but maybe we should go the other way and create a ticket to do some validation to make sure the subjects belong to the Release and all the other entities belong to those subjects?